centaur/src/it/scala/centaur/EngineUpgradeTestCaseSpec.scala

```diff
@@ -0,0 +1,75 @@
+package centaur
+
+import cats.effect.IO
+import centaur.EngineUpgradeTestCaseSpec._
+import cromwell.database.slick.{EngineSlickDatabase, MetadataSlickDatabase, SlickDatabase}
+import cromwell.database.sql.SqlDatabase
+import org.scalatest.{Assertions, BeforeAndAfter, DoNotDiscover}
+import shapeless.syntax.typeable._
+
+import scala.concurrent.Future
+
+@DoNotDiscover
+class EngineUpgradeTestCaseSpec(cromwellBackends: List[String])
+  extends AbstractCentaurTestCaseSpec(cromwellBackends)
+    with CentaurTestSuiteShutdown
+    with BeforeAndAfter {
+
+  def this() = this(CentaurTestSuite.cromwellBackends)
+
+  private val cromwellDatabase = CromwellDatabase.fromConfig(CentaurConfig.conf)
+  private val engineSlickDatabaseOption = cromwellDatabase.engineDatabase.cast[EngineSlickDatabase]
+  private val metadataSlickDatabaseOption = cromwellDatabase.metadataDatabase.cast[MetadataSlickDatabase]
+
+  override protected def beforeAll(): Unit = {
+    super.beforeAll()
+    val beforeAllIo = for {
+      _ <- checkIsEmpty(cromwellDatabase.engineDatabase, cromwellDatabase.engineDatabase.existsJobKeyValueEntries())
```

Comment:
Just OOC, why the check on job kv on top of metadata ?

Reply:
There can be two separate databases: "metadata" (to be removed from SQL one day) and "everything else" (aka "engine"). This does a cursory check to make sure both databases are empty.